crates/ruff_linter/src/rules/flake8_type_checking/helpers.rs

```diff
@@ -98,12 +98,15 @@ fn runtime_required_decorators(
     }
 
     decorator_list.iter().any(|decorator| {
+        let expression = map_callable(&decorator.expression);
         semantic
-            .resolve_qualified_name(map_callable(&decorator.expression))
+            .resolve_qualified_name(expression)
+            // if we can't resolve the name, then try resolving the assignment
+            .or_else(|| analyze::typing::resolve_assignment(expression, semantic))
```

Comment:
Nit: It might help future readers to add a small example here
```suggestion
						// ```python
						// from fastapi import FastAPI
						// @FastAPI.get
						// def test(): ...
						//  ```
            .resolve_qualified_name(expression)
            // if we can't resolve the name, then try resolving the assignment
						// ```python
						// from fastapi import FastAPI
						// app = FastAPI()
						// 
						// @app.get
						// def test(): ...
						// ```
            .or_else(|| analyze::typing::resolve_assignment(expression, semantic))
```